### PR TITLE
feature

### PR DESCRIPTION
**Pull request by TeodorTeaca**

### CODE REVIEW

File: src/app/list.movies/list.movies.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div>
+    <div *ngFor="let poster of posters">
+        <img src="http://image.tmdb.org/t/p/original/{{poster.posterPath}}" />
```
Comment (chitanumihai) [inline src/app/list.movies/list.movies.component.html:3]: refactor, map the object in the service

Comment (chitanumihai) [reply src/app/list.movies/list.movies.component.html:3]: rename the file movie-list.component.html

File: src/app/list.movies/list.movies.component.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  providers: [MoviesService]
+})
+
+export class ListMoviesComponent implements OnInit, OnDestroy {
+
+  constructor(private moviesService: MoviesService) { }
+
+  @Input() endpoint: string;
+  private request: Subscription;
+  public posters: object[];
```
Comment (chitanumihai) [inline src/app/list.movies/list.movies.component.ts:18]: add interface

Comment (chitanumihai) [reply src/app/list.movies/list.movies.component.ts:18]: or type

File: src/app/list.movies/list.movies.component.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  ngOnInit(): void {
+    console.log(this.endpoint);
+    this.request = this.moviesService.moviesRequest(this.endpoint)
+      .subscribe((posters) => {
```
Comment (chitanumihai) [inline src/app/list.movies/list.movies.component.ts:23]: handle error

File: src/app/services/movies.service.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { map } from 'rxjs/operators';
+
+@Injectable({ providedIn: 'root' })
+
+export class MoviesService {
+
+    constructor(private http: HttpClient) { }
+
+    moviesRequest(endPoint: string) {
+        return this.http.get(`https://api.themoviedb.org/3/movie/${endPoint}?api_key=8673e8c11eae0b8c433a41602fd2ddfc`)
```
Comment (chitanumihai) [inline src/app/services/movies.service.ts:12]: implement http interceptor, and extract url as const

File: src/app/services/movies.service.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+            .pipe(map((res: any) => res.results.map((movie: any) => ({ posterPath: `${movie.poster_path}` }))))
```
Comment (chitanumihai) [inline src/app/services/movies.service.ts:13]: implement interface

File: src/index.html [hunk 6]
```diff
@@ -1,13 +1,18 @@
 <!doctype html>
 <html lang="en">
+
 <head>
   <meta charset="utf-8">
   <title>AngularMovies</title>
   <base href="/">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
+  <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/css/bootstrap.min.css"
```
Comment (chitanumihai) [inline src/index.html:10]: this can be implemented in angular.json file

**Review by chitanumihai — CHANGES_REQUESTED** [review]
fix the comments

File: src/app/list.movies/list-movies.component.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  templateUrl: './list-movies.component.html',
+  styleUrls: ['./list-movies.component.css'],
+  providers: [MoviesService]
+})
+
+export class ListMoviesComponent implements OnInit, OnDestroy {
+  constructor(private moviesService: MoviesService) { }
+
+  @Input() endpoint: string;
+  public error: boolean = null;
```
Comment (chitanumihai) [inline src/app/list.movies/list-movies.component.ts:20]: ????

File: src/app/list.movies/list-movies.component.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+          this.error = error.error.status_message;
+        }
+      );
+  }
+
+  ngOnDestroy(): void {
+    this.request.unsubscribe();
+  }
+
+  private request: Subscription;
```
Comment (chitanumihai) [inline src/app/list.movies/list-movies.component.ts:40]: declarations are always before the constructor

**Review by chitanumihai — CHANGES_REQUESTED** [review]

File: src/app/list.movies/list-movies.component.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+  styleUrls: ['./list-movies.component.css'],
+  providers: [MoviesService]
+})
+
+export class ListMoviesComponent implements OnInit, OnDestroy {
+  constructor(private moviesService: MoviesService) { }
+
+  @Input() endpoint: string;
+  public error: boolean = null;
+  public posters: PosterList[];
```
Comment (vadimb) [inline src/app/list.movies/list-movies.component.ts:21]: : Poster[]

File: src/app/services/interceptor.service.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+import { LocalStorage } from '../services/local-storage.service';
+import { Injectable } from '@angular/core';
+
+@Injectable()
+
+export class InterceptorService implements HttpInterceptor {
+    constructor(private localStorage: LocalStorage) { }
+
+    intercept(req: HttpRequest<any>, next: HttpHandler) {
+        const session = this.localStorage.getElement('session');
```
Comment (vadimb) [inline src/app/services/interceptor.service.ts:12]: "Auth":

File: src/app/services/login.service.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+                    } else {
+                        throwError(error);
+                    }
+                })
+            )
+    }
+
+    getSessionRequest(token) {
+        return this.http.post(`${SESSION_PATH}`, { request_token: token })
+            .pipe(
```
Comment (vadimb) [inline src/app/services/login.service.ts:28]: {
REQUIRES_AUTH,
}

File: src/app/services/login.service.ts [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+export class LoginService {
+    constructor(private http: HttpClient) { }
+
+    getTokenRequest() {
+        return this.http.get(`${TOKEN_PATH}`)
+            .pipe(
+                map((res: any) => res.request_token),
+                catchError((error) => {
+                    if (error.stauts === 401) {
+                        return error.stauts;
```
Comment (vadimb) [inline src/app/services/login.service.ts:18]: throw(new AuthError(sfsdfsdf sfsdfs));

File: src/app/services/movies.service.ts [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+
+    moviesRequest(category: string) {
+        const path: string = `${MOVIE_PATH}${category}`;
+        return this.http.get(path)
+            .pipe(
+                map((res: PostersRes) => res.results
+                    .map((res) => res.poster_path)
+                    .map((res) => ({ posterPath: `${imageUrl}${res}` }))
+                ),
+                catchError((err) => throwError(err))
```
Comment (vadimb) [inline src/app/services/movies.service.ts:23]: ??????

File: src/app/services/movies.service.ts [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+export class MoviesService {
+    constructor(private http: HttpClient) { }
+
+    moviesRequest(category: string) {
+        const path: string = `${MOVIE_PATH}${category}`;
+        return this.http.get(path)
+            .pipe(
+                map((res: PostersRes) => res.results
+                    .map((res) => res.poster_path)
+                    .map((res) => ({ posterPath: `${imageUrl}${res}` }))
```
Comment (vadimb) [inline src/app/services/movies.service.ts:21]: Merge

**Pull request closed without merging**